pkg/component/etcd/poddisruptionbudget/valueshelper.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package poddisruptionbudget
+
+import (
+	"fmt"
+	"math"
+
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+)
+
+// GenerateValues generates `poddisruptionbudget.Values` for the lease component with the given parameters.
+func GenerateValues(etcd *druidv1alpha1.Etcd) Values {
+	labels := etcd.Spec.Labels
+	labels[instanceKey] = etcd.Name
+	labels[nameKey] = "etcd"
+	labels[appKey] = "etcd-statefulset"
+
+	selectorLabels := map[string]string{
+		instanceKey: etcd.Name,
+		nameKey:     "etcd",
+	}
+
+	annotations := map[string]string{
+		ownedByAnnotationKey:   fmt.Sprintf("%s/%s", etcd.Namespace, etcd.Name),
+		ownerTypeAnnotationKey: "etcd",
+	}
+
+	return Values{
+		EtcdName:       etcd.Name,
+		EtcdNameSpace:  etcd.Namespace,
+		EtcdUID:        etcd.UID,
+		Labels:         labels,
+		SelectorLabels: selectorLabels,
+		Annotations:    annotations,
+		MinAvailable:   int32(CalculatePDBMinAvailable(etcd)),
+	}
+}
+
+// CalculatePDBMinAvailable calculates the minimum available value for the PDB
+func CalculatePDBMinAvailable(etcd *druidv1alpha1.Etcd) int {
+	// do not enable in single mode
+	if etcd.Spec.Replicas < 2 {
+		return 0
+	}
+
+	clusterSize := int(etcd.Spec.Replicas)
+	return int(math.Floor(float64(clusterSize)/float64(2)) + 1)
```

Comment:
Why you are converting it to float, taking a floor and converting back to int.
you can directly use this: `clusterSize/2 + 1` as it is already int and int will truncate the result which is equivalent to taking a floor.

Reply:
Ah, thanks for pointing this out
This was a function that was already present in custodian_controller and I just moved it here, hence I didn't scrutinise it much
Updated